plugin/src/main/kotlin/spp/jetbrains/sourcemarker/status/LiveStatusManager.kt

```diff
@@ -74,6 +77,17 @@ object LiveStatusManager : SourceMarkEventListener {
 
     private val log = LoggerFactory.getLogger(LiveStatusManager::class.java)
     private val activeStatusBars = CopyOnWriteArrayList<LiveInstrument>()
+    private val logData = ConcurrentHashMap<String?, List<*>>()
```

Comment:
I think you should get rid of this `logData` map and put this in `SourceMarkKeys`.

The point of `SourceMarkKeys` is to associate custom data to invidivual `SourceMark`s (`InlayMark` is a type of `SourceMark`).

So the idea here would be that you could do something like:
```kt
inlayMark.getUserData(SourceMarkKeys.LOG_DATA)
```

Reply:
Tried this approach. Looks like the Inlaymark object reference is lost or cleared every time editor is closed so when editor reopened a new Inalymark instance is returned hence the user data is also lost. Keeping the old approach.

Reply:
Good catch